src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1346,7 +1348,7 @@ public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int
 
                 Debug.Assert(s_emptyTypeMembers.Count == 0);
                 return symbols.Count > 0 ?
-                    symbols.ToDictionary(s => s.Name, StringOrdinalComparer.Instance) :
+                    symbols.ToDictionary(s => s.Name.AsMemory(), ReadOnlyMemoryOfCharComparer.Instance) :
```

Comment:
ROM<char> does not have value-semantics (like `string` does).  So we pass this explicit comparer in all cases to ensure that lookups work properly.